jobs/pacman-awsrules/src/main/java/com/tmobile/cloud/awsrules/s3/EncryptS3BucketsAtRest.java

```diff
@@ -0,0 +1,117 @@
+	 * 
+	 * @param resourceAttributes this is a resource in context which needs to be scanned this is provided by execution engine
+	 *
+	 */
+
+	@Override
+	public RuleResult execute(final Map<String, String> ruleParam, Map<String, String> resourceAttributes) {
+
+		logger.debug("========EncryptS3BucketsAtRest started=========");
+
+		MDC.put("executionId", ruleParam.get("executionId"));
+		MDC.put("ruleId", ruleParam.get(PacmanSdkConstants.RULE_ID));
+
+		if (MapUtils.isNotEmpty(ruleParam) && !PacmanUtils.doesAllHaveValue(ruleParam.get(PacmanRuleConstants.SEVERITY),
+				ruleParam.get(PacmanRuleConstants.CATEGORY))) {
+			logger.info(PacmanRuleConstants.MISSING_CONFIGURATION);
+			throw new InvalidInputException(PacmanRuleConstants.MISSING_CONFIGURATION);
+		}
+
+		Optional<String> opt = Optional.ofNullable(resourceAttributes)
+				.map(resource -> checkValidation(ruleParam, resource));
+
+		RuleResult ruleResult = Optional.ofNullable(ruleParam).filter(param -> opt.isPresent())
+				.map(param -> buildFailureAnnotation(param, opt.get()))
+				.orElse(new RuleResult(PacmanSdkConstants.STATUS_SUCCESS, PacmanRuleConstants.SUCCESS_MESSAGE));
+
+		logger.debug("========EncryptS3BucketsAtRest ended=========");
+		return ruleResult;
+
+	}
+    
+	/**
+	 * @param ruleParam
+	 * @param resourceAttributes
+	 * @return
+	 * 
+	 * Check the ES attribute bucketencryp. If its empty indicates no encryption is enabled.
+	 * 
+	 */
+	private String checkValidation(Map<String, String> ruleParam, Map<String, String> resourceAttributes) {
+
+		String description = null;
+		String encryption = resourceAttributes.get(PacmanRuleConstants.ES_BKT_ENCRPTN_ATTRIBUTE);
+
+		if (StringUtils.isNullOrEmpty(encryption))
+			description = "S3 bucket without encryption found!!";
+		return description;
+	}
+
+	private static RuleResult buildFailureAnnotation(final Map<String, String> ruleParam, String description) {
+		
+		Annotation annotation = null;
+		LinkedHashMap<String, Object> issue = new LinkedHashMap<>();
+		List<LinkedHashMap<String, Object>> issueList = new ArrayList<>();
+
+		annotation = Annotation.buildAnnotation(ruleParam,Annotation.Type.ISSUE);
+		annotation.put(PacmanSdkConstants.DESCRIPTION,description);
+		annotation.put(PacmanRuleConstants.SEVERITY, ruleParam.get(PacmanRuleConstants.SEVERITY));
+		annotation.put(PacmanRuleConstants.CATEGORY, ruleParam.get(PacmanRuleConstants.CATEGORY));
+		annotation.put(PacmanRuleConstants.RESOURCE_ID, ruleParam.get(PacmanRuleConstants.RESOURCE_ID));
+		issue.put(PacmanRuleConstants.VIOLATION_REASON, description);
+		issueList.add(issue);
+		annotation.put("issueDetails",issueList.toString());
+		logger.debug("========EncryptS3BucketsAtRest annotation {} :=========",annotation);
+		return new RuleResult(PacmanSdkConstants.STATUS_FAILURE,PacmanRuleConstants.FAILURE_MESSAGE, annotation);
+	
+	
+	}
+	
+	@Override
+	public String getHelpText() {
+		return "Checks encryption at rest is enabled for S3 buckets";
+	}
```

Comment:
@amishavijayakumar08 , Are the rule config changes for this rule in cloudwatch_rules.json and sql changes in DB.sql not required?